### PR TITLE
Refactor AutoencoderKlMaisi

### LINKED ISSUE #7988: Refactor MAISI networks to use monai core components
**Is your feature request related to a problem? Please describe.**

[GenerativeModels](https://github.com/Project-MONAI/GenerativeModels) is moved to monai core. We plan to refactor three MAISI networks, including AutoencoderKlMaisi, DiffusionModelUNetMaisi, and ControlNetMaisi and use monai core components.

Public
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by guopengf**
Fixes #7988 .

### Description

Refactor AutoencoderKlMaisi to use monai core components.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by guopengf** [comment]
Hi @KumoLiu, would you like to provide instructions to solve this error?https://github.com/Project-MONAI/MONAI/actions/runs/10342336273/job/28625156457?pr=7993

**Comment by guopengf** [comment]
Hi @KumoLiu , would you like to provide instructions to solve this error? https://github.com/Project-MONAI/MONAI/actions/runs/10343670841/job/28628088785?pr=7993

**Comment by KumoLiu** [comment]
> Hi @KumoLiu , would you like to provide instructions to solve this error? https://github.com/Project-MONAI/MONAI/actions/runs/10343670841/job/28628088785?pr=7993

Hi @guopengf, add a commit to fix the mypy issue.

File: monai/apps/generation/maisi/networks/autoencoderkl_maisi.py [hunk 1]
```diff
@@ -523,6 +515,8 @@ class MaisiEncoder(nn.Module):
         norm_eps: Epsilon for the normalization.
         attention_levels: Indicate which level from num_channels contain an attention block.
         with_nonlocal_attn: If True, use non-local attention block.
+        include_fc: whether to include the final linear layer in the attention block. Default to True.
```
Comment (KumoLiu) [inline monai/apps/generation/maisi/networks/autoencoderkl_maisi.py:518]: Please also update the docstring for num_res_blocks.
`num_res_blocks: Number of residual blocks (see ResBlock) per level.`
to
`num_res_blocks: Number of residual blocks (see AEKLResBlock) per level.`

File: monai/apps/generation/maisi/networks/autoencoderkl_maisi.py [hunk 2]
```diff
@@ -704,6 +704,8 @@ class MaisiDecoder(nn.Module):
...
```
Comment (KumoLiu) [inline monai/apps/generation/maisi/networks/autoencoderkl_maisi.py:707]: Same as above for `num_res_blocks`

File: monai/apps/generation/maisi/networks/autoencoderkl_maisi.py [hunk 3]
```diff
@@ -886,6 +894,8 @@ class AutoencoderKlMaisi(AutoencoderKLType):
         norm_eps: Epsilon for the normalization.
         with_encoder_nonlocal_attn: If True, use non-local attention block in the encoder.
         with_decoder_nonlocal_attn: If True, use non-local attention block in the decoder.
+        include_fc: whether to include the final linear layer. Default to True.
```
Comment (KumoLiu) [inline monai/apps/generation/maisi/networks/autoencoderkl_maisi.py:897]:
```suggestion
        include_fc: whether to include the final linear layer. Default to False.
```

File: monai/apps/generation/maisi/networks/autoencoderkl_maisi.py (hunk 2 above)
Comment (KumoLiu) [inline monai/apps/generation/maisi/networks/autoencoderkl_maisi.py:707]:
```suggestion
        include_fc: whether to include the final linear layer in the attention block. Default to False.
```

File: monai/apps/generation/maisi/networks/autoencoderkl_maisi.py (hunk 1 above)
Comment (KumoLiu) [inline monai/apps/generation/maisi/networks/autoencoderkl_maisi.py:518]:
```suggestion
        include_fc: whether to include the final linear layer in the attention block. Default to False.
```

**Review by KumoLiu — APPROVED** [review]
Thanks for the PR, overall looks good to me.
Several minor comments regarding docstring, after you addressed, I will trigger more tests and merge this one.
Thanks.

File: monai/apps/generation/maisi/networks/autoencoderkl_maisi.py (hunk 1 above)
Comment (guopengf) [reply monai/apps/generation/maisi/networks/autoencoderkl_maisi.py]: Done

File: monai/apps/generation/maisi/networks/autoencoderkl_maisi.py (hunk 2 above)
Comment (guopengf) [reply monai/apps/generation/maisi/networks/autoencoderkl_maisi.py]: Done, thanks

**Pull request merged**